src/dmd/dinterpret.d

```diff
@@ -703,6 +715,13 @@ private Expression interpretFunction(UnionExp* pue, FuncDeclaration fd, InterSta
         e = CTFEExp.cantexp;
     }
 
+    // destroy temporaries
+    // https://issues.dlang.org/show_bug.cgi?id=22536
+    foreach(dtor; dtors)
+    {
+        auto res = interpret(dtor, istate);
```

Comment:
Why not ` evaluateDtor` which handles dtor calls?
Also `res` is unused and might contain a thrown exception.

Reply:
Because I am already checking if `edtor` (which is a CallExp) exists for the variables. It is wasteful to repeat the dtor checks that `evaluateDtor` does. 

I have fixed the exception part.